b2share/factory.py

```diff
@@ -1,192 +1,178 @@
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/communities', defaults={'path': ''})
-    @app_ui.route('/communities/', defaults={'path': ''})
-    @app_ui.route('/communities/<path:path>')
-    def serve_communities(path):
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/user', defaults={'path': ''})
-    @app_ui.route('/user/', defaults={'path': ''})
-    @app_ui.route('/user/<path:path>')
-    def serve_user(path):
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/records', defaults={'path': ''})
-    @app_ui.route('/records/', defaults={'path': ''})
-    @app_ui.route('/records/<path:path>')
-    def serve_records(path):
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/search', defaults={'path': ''})
-    @app_ui.route('/search/', defaults={'path': ''})
-    @app_ui.route('/search/<path:path>')
-    def serve_search(path):
-        return app_ui.send_static_file('index.html')
 
+    invenio_config_loader(app, **kwargs_config)
+
+
+class TrustedHostsMixin(object):
+    """Mixin for reading trusted hosts from application config."""
+
+    @property
+    def trusted_hosts(self):
+        """Get list of trusted hosts."""
+        if current_app:
+            return current_app.config.get('APP_ALLOWED_HOSTS', None)
+
+
+def app_class():
+    """Create Flask application class.
+
+    Invenio-Files-REST needs to patch the Werkzeug form parsing in order to
+    support streaming large file uploads. This is done by subclassing the Flask
+    application class.
+    """
+    try:
+        pkg_resources.get_distribution('invenio-files-rest')
+        from invenio_files_rest.app import Flask as FlaskBase
+    except pkg_resources.DistributionNotFound:
+        from flask import Flask as FlaskBase
+
+    # Add Host header validation via APP_ALLOWED_HOSTS configuration variable.
+    class Request(TrustedHostsMixin, FlaskBase.request_class):
+        pass
+
+    class Flask(FlaskBase):
+        request_class = Request
+
+    return Flask
+
+
+create_api = create_app_factory(
+    'invenio',
+    config_loader=config_loader,
+    blueprint_entry_points=['invenio_base.api_blueprints'],
+    extension_entry_points=['invenio_base.api_apps'],
+    converter_entry_points=['invenio_base.api_converters'],
+    wsgi_factory=wsgi_proxyfix(),
+    instance_path=instance_path,
+    app_class=app_class(),
+)
+"""Flask application factory for Invenio REST API."""
+
+#create_ui = create_app_factory(
+#    'invenio',
+#    config_loader=config_loader,
+#    blueprint_entry_points=['invenio_base.blueprints'],
+#    extension_entry_points=['invenio_base.apps'],
+#    converter_entry_points=['invenio_base.converters'],
+#    wsgi_factory=wsgi_proxyfix(),
+#    instance_path=instance_path,
+#    static_folder=static_folder,
+#    static_url_path=static_url_path(),
+#    app_class=app_class(),
+#)
+#"""Flask application factory for Invenio UI."""
```

Comment:
This implementation was commented out because it created problems with unit_tests.
**DO NOT REMOVE** Since it is related to implementation below (`create_app`)